### PR TITLE
spirv-val: Add SPV_KHR_ray_tracing instructions

### PR DESCRIPTION
**Pull request by sjfricke**
(tasklist https://github.com/KhronosGroup/Vulkan-Docs/issues/1402)

### CODE REVIEW

File: source/val/validate_ray_tracing.cpp [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+      const uint32_t hit_kind = _.GetOperandTypeId(inst, 3);
+      if (!_.IsUnsignedIntScalarType(hit_kind) ||
+          _.GetBitWidth(hit_kind) != 32) {
+        return _.diag(SPV_ERROR_INVALID_DATA, inst)
+               << "Hit Kind must be a 32-bit unsigned int scalar";
+      }
+      break;
+    }
+
+    case SpvOpExecuteCallableKHR: {
```
Comment (alan-baker) [inline source/val/validate_ray_tracing.cpp:159]: SBT index? Spec doesn't really say much, but certainly it should be an integer. Not sure if it is restricted to 32 bits though.

Comment (sjfricke) [reply source/val/validate_ray_tracing.cpp:159]: Ya, I noticed that already and made this with the PR
https://github.com/KhronosGroup/SPIRV-Registry/issues/156

I was going to wait for the WG to just confirm/respond and then add it in `spirv-val`

Comment (alan-baker) [reply source/val/validate_ray_tracing.cpp:159]: That seems reasonable.

**Review by alan-baker — APPROVED** [review]

**Comment by alan-baker** [comment]
Not sure why this formatting issue is coming up for you. Mind rebasing?

File: source/val/validate_ray_tracing.cpp (hunk 1 above)
Comment (sjfricke) [reply source/val/validate_ray_tracing.cpp:159]: https://github.com/KhronosGroup/SPIRV-Registry/pull/160 - will make new PR to add it

**Pull request merged**